File: MovingStone/stone.py
while True:
    # Read input
    n = input()
    if n < 0:
        break

    a = []
    while len(a) < n:
        a.extend(map(int, raw_input().split()))

    # total = number of stones
    total = sum(a)

    # res = our final result
    res = 0
    if total % n > 0:
        res = -1
    else:
        each = total / n  # How many stones each box have at final
        l = 0
        r = 0

        for i in xrange(n):
            a[i] += r - l

            if a[i] > each:
                # If a[i] > each, we must move stones to our right
                l = 0
                r = a[i] - each
            else:
                # Otherwise, we must borrow from our right
                l = each - a[i]
                r = 0

            # Update result
            res = max(res, max(l, r))

    raw_input()
    print res

File: MovingStone/stone.java
import java.io.*;
import java.math.*;
import java.util.*;

public class stone {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        while (true) {
            int n = sc.nextInt();
            if (n < 0) break;

            long[] a = new long[n];
            long total = 0;
            for(int i = 0; i < n; ++i) {
                a[i] = sc.nextLong();
                total += a[i];
            }

            long res = 0;
            if (total % n == 0) {
                long each = total / n;
                long l = 0, r = 0;
                for(int i = 0; i < n; ++i) {
                    a[i] += r - l;
                    if (a[i] > each) {
                        l = 0;
                        r = a[i] - each;
                    }
                    else {
                        l = each - a[i];
                        r = 0;
                    }

                    res = Math.max(res, Math.max(l, r));
                }
            }
            else res = -1;
            System.out.println(res);
        }
        sc.close();
    }
}
